### PR TITLE
[rpc] default construct rpc agent options based on the backend type

### PR DESCRIPTION
**Pull request by rohan-varma**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30208 [rpc] add default arg for init_method
* **#30201 [rpc] default construct rpc agent options based on the backend type**

Provide a default constructor so that users don't have to construct
RPC agent options. Also rename this to RPCBackend Options as suggested.

Differential Revision: [D18628698](https://our.internmc.facebook.com/intern/diff/D18628698/)

### CODE REVIEW

File: torch/distributed/rpc/api.py [hunk 1]
```diff
@@ -74,6 +74,12 @@ def _init_rpc_backend(
 ):
     from . import RpcAgentOptions
 
+    if not rpc_agent_options:
+        # default construct a set of RPC agent options.
+        rpc_agent_options = rpc.backend_registry.construct_rpc_agent_options(
+            backend
+        )
```
Comment (xush6528) [inline torch/distributed/rpc/api.py]: Make sense.

construct_rpc_agent_options handlers should provide defaults in Python world.

**Review by xush6528 — APPROVED** [review]

**Review by xush6528 — CHANGES_REQUESTED** [review]
Noticed your PR summary and you want to do the renaming as well.

**Review by xush6528 — APPROVED** [review]
Thanks, Rohan!

File: torch/distributed/rpc/api.py [hunk 2]
```diff
@@ -70,35 +76,20 @@ def _init_rpc_backend(
...
-        raise RuntimeError("`world_size` must be an integer. {}".format(world_size))
-
-    if not isinstance(rpc_agent_options, RpcAgentOptions):
-        raise RuntimeError(
-            "`rpc_agent_options` must be an `RpcAgentOptions`. {}".format(
-                rpc_agent_options
-            )
+    if not rpc_backend_options:
+        # default construct a set of RPC agent options.
+        rpc_backend_options = rpc.backend_registry.construct_rpc_backend_options(
```
Comment (mrshenli) [inline torch/distributed/rpc/api.py]: nice!

**Pull request closed without merging**